### PR TITLE
Note that yarn dev needs webserver in debug mode

### PR DESCRIPTION
**Pull request by bbovenzi**
Some contributors have been confused on how to iterate quickly with the UI. Often building the UI after every change. But `airflow webserver -d` and `yarn dev` allows for just refreshing the page.
This PR adds a note to the contributing doc to help surface this information for future contributors.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by jedcunningham — APPROVED** [review]

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -1253,6 +1253,7 @@ commands:
     yarn run prod
 
     # Starts a web server that manages and updates your assets as you modify them
+    # You'll need to run the webserver in dev mode too: `airflow webserver -d`
```
Comment (jedcunningham) [inline CONTRIBUTING.rst]:
```suggestion
    # You'll need to run the webserver in debug mode too: `airflow webserver -d`
```

Let's use the language the webserver uses.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -1253,6 +1253,7 @@ commands:
     yarn run prod
 
     # Starts a web server that manages and updates your assets as you modify them
+    # You'll need to run the webserver in debug mode too: `airflow webserver -d`
```
Comment (eladkal) [inline CONTRIBUTING.rst:1256]: does `-d` work?
isn't it `-D` or `--daemon`?
https://github.com/apache/airflow/blob/3138604b264878f27505223bd14c7814eacc1e57/airflow/cli/cli_parser.py#L213

Comment (jedcunningham) [reply CONTRIBUTING.rst:1256]: We are after debug, which is `-d`.
https://github.com/apache/airflow/blob/3138604b264878f27505223bd14c7814eacc1e57/airflow/cli/cli_parser.py#L608-L610

**Review by jedcunningham — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

File: CONTRIBUTING.rst (hunk 2 above)
Comment (bbovenzi) [reply CONTRIBUTING.rst:1256]: I've always been using `-d` so I gets its not case specific. But our docs do use `-D`, so I'll make it consistent

Comment (jedcunningham) [reply CONTRIBUTING.rst:1256]: `-D` is different! You want the lower case `-d`. Daemon vs debug.

Comment (bbovenzi) [reply CONTRIBUTING.rst:1256]: Reverting.

**Pull request merged**